query/functions.gen.go.tmpl

```diff
@@ -1,37 +1,27 @@
 package query
 
 import (
+"encoding/binary"
+"bytes"
 "sort"
 "time"
 "math/rand"
+
+"github.com/influxdata/influxdb/pkg/estimator/hll"
 )
 
 {{with $types := .}}{{range $k := $types}}
 
 // {{$k.Name}}PointAggregator aggregates points to produce a single point.
 type {{$k.Name}}PointAggregator interface {
-	Aggregate{{$k.Name}}(p *{{$k.Name}}Point)
+	Aggregate{{$k.Name}}(p *{{$k.Name}}Point) error
```

Comment:
This interface change is in a separate commit - we need to decide if it is a good idea.

Reply:
@dgnorton and I talked and we agree that this seems like a good idea.

Reply:
Does this change have a noticeable impact on performance? It is unclear to me how frequently this API is called, however, if large enough, pushing an extra value (the `error` return) on to the stack may have an impact.

Reply:
I added BenchmarkIterator_Aggregator and ran several times (see https://gist.github.com/lesam/9337ce60306d638f03a6bfcf17c42a32 for methodology)

```
benchstat without-error-aggregate.txt with-error-aggregate.txt
name                    old time/op  new time/op  delta
Iterator_Aggregator-12  26.4ns ± 7%  26.7ns ± 1%  +1.23%  (p=0.006 n=19+17)
```

Interpretation: This is about 1% worse performance in the absolute worst case (counting a const value). I would expected real-world `count` to have to do more work (e.g. actually writing to the point) so in the real world I would expect less than 1% impact. I think it's worth it for the code cleanliness, but I can fake the error return with a different string if anyone disagrees.

Reply:
Thinking about it more, 1% regression is probably enough to change this around, since only count_hll needs it - let me rework this a bit.

Reply:
> ...in the real world I would expect less than 1% impact

I don't know if < 1% real world impact is worth it. Passing an error via a string will make for confusing code.

Reply:
The re-write turned out to be pretty understandable and actually a lot less change overall, I think it should be ok.